### DISCUSSION TITLE
BFD exporter question

### DISCUSSION THREAD

**Discussion by saywurdson** [synthetichealth/synthea#1443] [Q&A] [answered]
Hi @hadleynet, I just tried to follow your instructions (see referenced issue below) and am running into an error with the following .csv files:
external_codes.csv
hha_pps_case_mix_codes.csv
hha_pps_pdgm_codes.csv

my goal is to generate this missing_codes.csv to build the mapping files. Can I get some help figuring out the structure for these files so that I can bypass this error?

here is the referenced discussion:

#1389 
https://github.com/synthetichealth/synthea/discussions/1389
I'm interested in using the BFD export option to get output that looks like [Medicare's Synthetic RIF files](https://data.cms.gov/sites/default/files/2023-05/d51e1218-68c3-4c7c-9598-0b81f22fe903/User%20Guide%20-%20CMS%20Synthetic%20RIF%20Files%20May%202023_AM508_v2.pdf).

I was hoping it'd be as easy as just setting line 60 of the synthea.properties file to True, but it looks like I need to provide a bunch of code maps in .json format (eg "export/drg_code_map.json" and "export/hcpcs_code_map.json").

Am I right in guessing that you do not provide these because of licensing issues? If we can get our own licenses, would we be able to request these files or create them ourselves? What is the general format and content of these code map files?

Originally posted by @rileeki in https://github.com/synthetichealth/synthea/discussions/1389

**Comment by TravisHaussler** [comment]
@saywurdson were you able to overcome your error ever?  I would be interested if you had success here!

**Reply by curikwasi** [reply]
@TravisHaussler Not yet.

I'm really just trying to find out what the format is for those 3 files (what columns are in the files, etc). I'm hoping that if I can get that format, I can just put blank csvs in the project and generate that missing_codes.csv and then go from there

**Reply by jawalonoski** [reply, marked as answer]
If you just want it to run, set `--exporter.bfd.require_code_maps=false` on the command line or in the `synthea.properties` file.

Here is the format of those files.

## external_codes.csv

Column | Type 
--------|-------
`primary` | String
`external` | String
`count` | Integer

## hha_pps_case_mix_codes.csv

Column | Type 
--------|-------
`code` | String
`count` | Integer

## hha_pps_pdgm_codes.csv

Column | Type 
--------|-------
`code` | String
`count` | Integer
